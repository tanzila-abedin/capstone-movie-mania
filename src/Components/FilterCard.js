import PropTypes from 'prop-types';

const FilterCard = ({ title, handleFilter }) => (
  <>
    <div className="mt-5 text-center">
      <div className="container pb-5 form-bg">
        <h1 className="display-4 mb-3 text-white" data-testid="filterCard">
          {' '}
          Search for a popular movie
        </h1>
        <form id="searchForm">
          <input
            type="search"
            value={title}
            placeholder="start typing to filter.."
            onChange={handleFilter}
            className="form-control"
          />
        </form>
      </div>
    </div>
  </>
);

FilterCard.propTypes = {
  title: PropTypes.string.isRequired,
  handleFilter: PropTypes.func.isRequired,
};

export default FilterCard;
