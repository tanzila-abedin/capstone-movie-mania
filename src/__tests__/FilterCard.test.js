import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import FilterCard from '../Components/FilterCard';

test('Displays heading on filter card component', async () => {
  const title = 'string';
  const handleFilter = jest.fn();
  const home = render(
    <FilterCard title={title} handleFilter={handleFilter} />,
  );
  const welcomeText = await home.findByTestId('filterCard');
  expect(welcomeText.textContent).toBe(' Search for a popular movie');
});

test('FilterCard component is defined', async () => {
  const title = 'string';
  const handleFilter = jest.fn();
  const component = render(
    <FilterCard title={title} handleFilter={handleFilter} />,
  );
  const filter = await component.findAllByTestId('filterCard');
  expect(filter).toBeDefined();
});

test('FilterCard snapshot with react and redux  ', () => {
  const title = 'string';
  const handleFilter = jest.fn();
  const app = renderer.create(
    <FilterCard title={title} handleFilter={handleFilter} />,
  ).toJSON();
  expect(app).toMatchSnapshot();
});
