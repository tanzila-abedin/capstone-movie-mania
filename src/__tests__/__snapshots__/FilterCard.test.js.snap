// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterCard snapshot with react and redux   1`] = `
<div
  className="mt-5 text-center"
>
  <div
    className="container pb-5 form-bg"
  >
    <h1
      className="display-4 mb-3 text-white"
      data-testid="filterCard"
    >
       
      Search for a popular movie
    </h1>
    <form
      id="searchForm"
    >
      <input
        className="form-control"
        onChange={[MockFunction]}
        placeholder="start typing to filter.."
        type="search"
        value="string"
      />
    </form>
  </div>
</div>
`;
